org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/XtextProjectCreator.java

```diff
@@ -125,6 +127,11 @@ private void configureJavaProject(ProjectDescriptor descriptor, JavaProjectFacto
 		if (needsBuildshipIntegration(descriptor) && !descriptor.isEclipsePluginProject()) {
 			factory.addClasspathEntries(JavaCore.newContainerEntry(new Path("org.eclipse.buildship.core.gradleclasspathcontainer")));
 		}
+		if (descriptor instanceof TestProjectDescriptor) {
+			JUnitVersion junitVersion = descriptor.getConfig().getJunitVersion();
```

Comment:
Do we need this in grade and pure maven too

Reply:
Yes. The projects only have a dependency on the JU API, not on the test engines. To execute the tests in Eclipse, we still need the JU container.

Reply:
It might be necessary to add the engine dependencies for the surefire plugin, but the project itself should not depend on them for similar reason as in Tycho.

Reply:
Ok should be handle this with a different pr / the enhance Junit 5 support with 2.15 issue

Reply:
At least with different PR. If possible for 2.14, I'll try that.